kliff/transforms/property_transforms.py

```diff
@@ -0,0 +1,189 @@
+from typing import TYPE_CHECKING, List, Union
+
+import numpy as np
+
+# if TYPE_CHECKING:
+from kliff.dataset import Configuration, Dataset
+
+
+class PropertyTransform:
+    """
+    A property transform is a function that maps a property of a configuration to a
+    transformed property. For example, current property transforms include normalization
+    of the energy and forces.
+
+    Attributes:
+        keep_original: If True, the original property values are kept in the configuration
+            object. Otherwise, the original property values are discarded.
+        original_property_value_map: A map that stores the original property values of
```

Comment:
I have no issues with it. But that would mean all the property transforms need to be bijective. The main example I had in mind was ML models to predict norm of forces, so that scalar values can be predicted per atom-pair, which can then be used to generate force pairs per vector. Leaving this unresolved for now. Will remove if needed in next iteration.

Reply:
right, I was only thinking of the bijective case. Then we should keep it.